helix-term/src/commands.rs

```diff
@@ -4746,32 +4783,38 @@ fn replace_with_yanked(cx: &mut Context) {
 }
 
 fn replace_with_yanked_impl(editor: &mut Editor, register: char, count: usize) {
-    let Some(values) = editor
-        .registers
-        .read(register, editor)
-        .filter(|values| values.len() > 0)
-    else {
+    let scrolloff = editor.config().scrolloff;
+
+    let Some(values) = editor.registers.read(register, editor) else {
         return;
     };
-    let scrolloff = editor.config().scrolloff;
-    let (view, doc) = current_ref!(editor);
+    let yanked = values.map(|value| value.to_string()).collect::<Vec<_>>();
+    let (view, doc) = current!(editor);
 
-    let map_value = |value: &Cow<str>| {
+    replace_impl(&yanked, doc, view, scrolloff, count)
+}
+
+fn replace_impl(
+    values: &[String],
+    doc: &mut Document,
+    view: &mut View,
+    count: usize,
+    scrolloff: usize,
+) {
+    let map_value = |value: &String| {
         let value = LINE_ENDING_REGEX.replace_all(value, doc.line_ending.as_str());
         let mut out = Tendril::from(value.as_ref());
         for _ in 1..count {
             out.push_str(&value);
         }
         out
     };
-    let mut values_rev = values.rev().peekable();
-    // `values` is asserted to have at least one entry above.
-    let last = values_rev.peek().unwrap();
+    let mut values_rev = values.iter().rev().peekable();
+    let Some(last) = values_rev.peek() else {
```

Comment:
Why did you change this from an unwrap? As far as I can see there is still always atleast one value? 

And if there isn't I would rather unwrap instead of silently doing nothing (or atleast log an error)